### PR TITLE
Update profile web overlay to use `SoloScoreInfo`

### LINKED ISSUE #19146: Score's dates in browser has been reset to 1 January of 0001 year
### Type

Game behaviour

### Bug description

2022 years ago

### Screenshots or videos

image

### Version

716.1

### Logs

[network.log](https://github.com/ppy/osu/files/9125726/network.log)
[runtime.log](https://github.com/ppy/osu/files/9125727/runtime.log)
[performance.log](https://github.com/ppy/osu/files/9125728/performance.log)
[database.log](https://github.com/ppy/osu/files/9125729/database.log)
[updater.log](https://github.com/ppy/osu/files/9125730/updater.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/9125731/legacy-ipc.log)

### LINKED ISSUE #19152: "Ranks" section of profile doesn't list plays
### Type

Game behaviour

### Bug description

Happens on [my in-game profile](https://osu.ppy.sh/users/8543042). I don't know if it is related to #19146 since it only seems to happen in the "Ranks" section.
The `network.log` seems to contain the problem (see below).
**I also wrote a message in the `#lazer` chat to quickly access the profile if you need it.**

### Screenshots or videos

image

### Version

2022.716.1-lazer

### Logs

[network.log](https://github.com/ppy/osu/files/9125870/network.log)

### PR DESCRIPTION
**Pull request by tsunyoku**
This introduces a new paginated class to handling the retrieval of scores in the new format, aswell as their beatmaps. This also replaces the use of `APIScore` in `GetUserScoresRequest` with `SoloScoreInfo` instead, and updates the relevant tests.

Resolves https://github.com/ppy/osu/issues/19152
Resolves https://github.com/ppy/osu/issues/19146

### CODE REVIEW

**Comment by tsunyoku** [comment]
Turns out a lot of this is unneeded since `SoloScoreInfo` should have `Beatmap`. Will adjust the PR, and then it should be ready.

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ private void load(RulesetStore rulesets)
                                                         Font = OsuFont.GetFont(size: 12, weight: FontWeight.Regular),
                                                         Colour = colours.Yellow
                                                     },
-                                                    new DrawableDate(Score.Date, 12)
+                                                    new DrawableDate(Score.EndedAt ?? DateTimeOffset.Now, 12)
```
Comment (frenzibyte) [inline osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:101]: `DateTimeOffset.Now` isn't really a good fall back if the score can't provide the date. At the very least it should not add a drawable date at all if missing.

**Review by frenzibyte — CHANGES_REQUESTED** [review]
Have pushed a change to allow mods displayed in profile scores to include settings as well (by using `APIMod.ToMod` instead of `CreateModFromAcronym(APIMod.Acronym`).

Seems fine otherwise upon a closer look at the response returned with the old and new API structures.

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs (hunk 1 above)
Comment (tsunyoku) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:101]: I did this match behaviour in another place where `SoloScoreInfo` is used. I don’t remember where but I did it to remain consistent. I can change it if you want.

Comment (frenzibyte) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:101]: It is used in `SoloScoreInfo.ToScoreInfo`, correct, but it doesn't seem like that method had any better way to handle null `ended_at` value.

Comment (peppy) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableProfileScore.cs:101]: I've made this change in #19210. Seems to be okay.

**Review by frenzibyte — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**